components/os/scripts/setup-ssh.ps1

```
@@ -3,22 +3,34 @@ $service = Get-Service -Name sshd -ErrorAction SilentlyContinue
 if ($service -ne $null) {
   Write-Host "Stop sshd service"
   Stop-Service sshd
-} 
-
-if (-not (Get-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0  | Where-Object { $_.State -eq 'Installed' })) {
-  Write-Host "Add OpenSSH Server capability"
-  Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0
 } else {
-  Write-Host "OpenSSH Server capability already installed"
+  Write-Host "sshd service not found, installing OpenSSH Server"
+    # Add-WindowsCapability does NOT install a consistent version across Windows versions, this lead to
+    # compatability issues (different command line quoting rules).
+    # Prefer installing sshd via MSI  
+    start-process -passthru -wait msiexec.exe -args '/i https://github.com/PowerShell/Win32-OpenSSH/releases/download/v9.5.0.0p1-Beta/OpenSSH-Win64-v9.5.0.0.msi /qn'
```

do you want to fail early if SSH fails to install? if so check the exit code.

```suggestion
    $p=start-process -passthru -wait msiexec.exe -args '/i https://github.com/PowerShell/Win32-OpenSSH/releases/download/v9.5.0.0p1-Beta/OpenSSH-Win64-v9.5.0.0.msi /qn'
    if ($p.ExitCode -ne 0) {
        throw "SSH install failed: $($p.ExitCode)"
    }
```